swap_meet/vendor.py

```diff
@@ -1,2 +1,74 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        inventory = [] if inventory is None else inventory
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+   
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.add(their_item)
+        self.remove(my_item)
+
+        other_vendor.add(my_item)
+        other_vendor.remove(their_item)
+        return True
+    
+    def swap_first_item(self,other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        our_first_item = self.inventory[0]
+        friend_first_item = other_vendor.inventory[0]
+
+        self.swap_items(other_vendor, our_first_item, friend_first_item)
+        return True
+    
+    def get_by_category(self,given_category):
+        item_by_category = []
+        for item in self.inventory:
+            if item.category == given_category:
+                item_by_category.append(item)
+        return item_by_category
+    
+    
+    def get_best_by_category(self, given_category):
+        items_in_category_list = self.get_by_category(given_category)
+        if not items_in_category_list:
+            return None
+        
+        return max(items_in_category_list, key = lambda item: item.condition)
```

Comment:
Omg, yes! Great work with the `lambda` function! This was a perfect use case scenario!